### PR TITLE
[ESIMD] Enable math functions: cos,sin,exp,log

### PR DESCRIPTION
**Pull request by DenisBakhvalov**

### CODE REVIEW

File: SYCL/ESIMD/ext_math.cpp [hunk 1]
```diff
@@ -159,13 +178,10 @@ template <int VL> bool test(queue &Q) {
   Pass &= test<MathOp::sqrt, VL>(Q, "sqrt", InitDataFuncWide{});
   Pass &= test<MathOp::inv, VL>(Q, "inv");
   Pass &= test<MathOp::rsqrt, VL>(Q, "rsqrt");
-// TODO enable these tests after the implementation is fixed
-#if ENABLE_SIN_COS_EXP_LOG
   Pass &= test<MathOp::sin, VL>(Q, "sin", InitDataFuncWide{});
   Pass &= test<MathOp::cos, VL>(Q, "cos", InitDataFuncWide{});
-  Pass &= test<MathOp::exp, VL>(Q, "exp");
+  Pass &= test<MathOp::exp, VL>(Q, "exp", InitDataInRange0_5{});
```
Comment (DenisBakhvalov) [inline SYCL/ESIMD/ext_math.cpp:185]: The precision of FP operations gets lower towards bigger numbers, that's why we use the smaller value range.

File: SYCL/ESIMD/ext_math.cpp [hunk 2]
```diff
@@ -61,10 +71,19 @@ template <MathOp Op> float HostMathFunc(float X);
     }                                                                          \
   }
 
-DEFINE_OP(sin, sin);
-DEFINE_OP(cos, cos);
-DEFINE_OP(exp, exp);
-DEFINE_OP(log, log);
+#define DEFINE_OP_REUSE_SCALAR(Op, HostOp)                                     \
```
Comment (kbobrovs) [inline SYCL/ESIMD/ext_math.cpp]: I think it is worth to clearly state that this is std sycl:
```suggestion
#define DEFINE_SIMD_OVERLOADED_STD_SYCL_OP(Op, HostOp)                                     \
```

File: SYCL/ESIMD/ext_math.cpp [hunk 3]
```diff
@@ -61,10 +71,19 @@ template <MathOp Op> float HostMathFunc(float X);
...
+    operator()(const simd<float, VL> &X) const SYCL_ESIMD_FUNCTION {           \
+      return sycl::Op<VL>(X);                                                  \
+    }                                                                          \
+  }
+
+DEFINE_OP_REUSE_SCALAR(sin, sin);
+DEFINE_OP_REUSE_SCALAR(cos, cos);
+DEFINE_OP_REUSE_SCALAR(exp, exp);
+DEFINE_OP_REUSE_SCALAR(log, log);
 DEFINE_OP(inv, 1.0f /);
```
Comment (kbobrovs) [inline SYCL/ESIMD/ext_math.cpp]:
```suggestion
DEFINE_ESIMD_OP(inv, 1.0f /);
```

Comment (DenisBakhvalov) [reply SYCL/ESIMD/ext_math.cpp]: Done

File: SYCL/ESIMD/ext_math.cpp (hunk 2 above)
Comment (DenisBakhvalov) [reply SYCL/ESIMD/ext_math.cpp]: Done

**Review by kbobrovs — DISMISSED** [review]

**Comment by DenisBakhvalov** [comment]
We need to wait until the next GPU driver uplift happens. The test will pass starting from the 21.24.20098 version.

**Comment by vladimirlaz** [comment]
The 21.24.20098 driver is going to be used since 02/30/2021 nightly

**Comment by vladimirlaz** [comment]
@DenisBakhvalov it looks like the modified test is failing on Windows. Could you please address the issue?

**Comment by DenisBakhvalov** [comment]
@kbobrovs, please review/approve.

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**